### DISCUSSION TITLE
[Best practice] CNF's containers should run unprivileged

### DISCUSSION THREAD

**Discussion by taylor** [lfn-cnti/bestpractices#25] [Ideas]
### References

- https://x.cnf.dev/process-containers/
- https://www.cncf.io/blog/2020/10/16/hack-my-mis-configured-kubernetes-privileged-pods/
- https://nirmata.com/2020/02/19/10-kubernetes-best-practices-you-can-easily-apply-to-your-clusters/
- https://kubernetes.io/blog/2018/07/18/11-ways-not-to-get-hacked/
- https://cloud.google.com/solutions/best-practices-for-operating-containers#avoid_privileged_containers

**Comment by taylor** [comment]
@iawells: Exceptions should be documented

@fkautz:
CNFs should follow the [Principle of Least Privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)

**Reply by jeffsaelens** [reply]
We also need to provide context for the type of container this practice applies to.

**Reply by taylor** [reply]
See related higher-level discussion on the principle of least privilege in https://github.com/cncf/cnf-wg/discussions/28

**Comment by cristicalin** [comment]
How do we handle workloads that need privileges to interact with devices? Think about SR-IOV or DPDK, in these cases you need root and cap=SYS_ADMIN|NET_ADMIN to interact with these.

Do we need additional primitives to describe behaviors like acquiring initial resources then dropping privileges?

Do we have a better way of doing this on behalf of the CNF in a way that the privileged work is separated from the non-privileged work and locked behind proper security mechanisms?
